### PR TITLE
MULE-19169: FirstSuccessful is not working inside ParallelForeach 

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: integration/src/test/java/org/mule/test/routing/FirstSuccessfulTestCase.java [hunk 1]
```diff
@@ -98,4 +98,10 @@ public void insideErrorHandlerFailing() throws Exception {
     flowRunner("firstSuccessfulInErrorHandlerWithFailing").withPayload(Boolean.TRUE).run().getMessage();
   }
 
+  @Test
+  public void firstSuccessfulInsideParalleLForEach() throws Exception {
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/routing/FirstSuccessfulTestCase.java:102]: `@Issue`

File: integration/src/test/resources/first-successful-test.xml [hunk 2]
```diff
@@ -126,4 +126,16 @@
             </on-error-continue>
         </error-handler>
     </flow>
+
+    <flow name="firstSuccessfulInParallelForEach">
+        <set-payload value="#[[1, 2]]"/>
+        <parallel-foreach>
+            <first-successful>
+                <route>
+                    <test:processor processingType="CPU_INTENSIVE"/>
```
Comment (elrodro83) [inline integration/src/test/resources/first-successful-test.xml:135]: can you use a queue here to also assert that this route is executed?

File: integration/src/test/resources/first-successful-test.xml [hunk 3]
```diff
@@ -126,4 +126,16 @@
             </on-error-continue>
         </error-handler>
     </flow>
+
+    <flow name="firstSuccessfulInParallelForEach">
+        <set-payload value="#[[1, 2]]"/>
+        <parallel-foreach>
+            <first-successful>
+                <route>
```
Comment (elrodro83) [inline integration/src/test/resources/first-successful-test.xml:134]: is a first-successful with just one route actually valid?

Comment (elrodro83) [reply integration/src/test/resources/first-successful-test.xml:134]: add a TODO to validate this

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: integration/src/test/resources/first-successful-test.xml (hunk 3 above)
Comment (mbuchwald) [reply integration/src/test/resources/first-successful-test.xml:134]: It is valid, probably makes no sense but there is no place in which we say or validate differently

Comment (mbuchwald) [reply integration/src/test/resources/first-successful-test.xml:134]: (including xml or docs)

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**